protoutil/blockutils.go

```diff
@@ -298,3 +299,35 @@ func searchConsenterIdentityByID(consenters []*cb.Consenter, identifier uint32)
 	}
 	return nil
 }
+
+func VerifyTransactionsAreWellFormed(block *cb.Block) error {
+	if block == nil || block.Data == nil {
+		return nil
+	}
+
+	for i, rawTx := range block.Data.Data {
+		env := &cb.Envelope{}
+		if err := proto.Unmarshal(rawTx, env); err != nil {
+			return fmt.Errorf("transaction %d is invalid: %v", i, err)
+		}
+
+		if len(env.Payload) == 0 {
+			return fmt.Errorf("transaction %d has no payload", i)
+		}
+
+		if len(env.Signature) == 0 {
+			return fmt.Errorf("transaction %d has no signature", i)
+		}
+
+		expected := MarshalOrPanic(env)
```

Comment:
The input to this loop is a rawTx supplied by a foreign process that may be malicious.
It is therefore not good form to throw panic in response to anything that comes from a foreign party.
Here the risk is low, as there are very few cases in proto3  in which proto.Marshal a message causes an error. This however may change in the future depending on the message and implementation. For example in proto3 a missing required field in the message would cause that (but we are proto3). 
In the code base MarshalOrPanic is used mainly in tests, or when the same process produced the message and now marshals it. 
This is a nit, but in my opinion, since it is so easy to fix, I would do it.